backend/local/backend_plan.go

```diff
@@ -263,7 +263,7 @@ func RenderPlan(plan *plans.Plan, state *states.State, schemas *terraform.Schema
 			continue
 		}
 
-		// FIXME currently only available in config - where should this live?
+		// FIXME: FQN needs to be recorded in both state and plan
```

Comment:
This looks like an example of what I was fearing about `AbsProviderConfig` not containing the FQN "natively" and thus we start needing to refer to the configuration in a lot more places.

I think if we are able to deal with `addrs.AbsProviderConfig` not embedding `addrs.LocalProviderConfig` (which the idea in my previous comment hopefully enables) then we should be able to say just `providerFqn := rcs.ProviderAddr.Provider` here, right? `rcs.ProviderAddr` would record the FQN we determined during planning, avoiding the need for us to re-resolve it from the configuration again.

Reply:
I had planned on _something_ along the lines of adding a `ProviderFqn` directly to the `rcs`. My goal was not to change plan and state at all in this PR and just leave placeholders (like this unfortunate comment) for that upcoming work.